### PR TITLE
Fix search images

### LINKED ISSUE #78: Implementing the search function
The `datataset.search()` function at present is not really implemented.

We are looking to implement it, thinking what would be a good output of this - a quick solution could be just a list of `Image` instances, which contains reference to filenames and relevant annotations.

Keen to hear others thoughts on this and what would you use the function for.

And if anyone is willing to contribute to the codebase, that's welcome too!

### PR DESCRIPTION
**Pull request by vmanilo**
Closes #78

### CODE REVIEW

File: remo/domain/image.py [hunk 1]
```diff
@@ -147,3 +166,12 @@ def view_annotate(self, annotation_set_id: int):
             annotation_set_id: annotation set id
         """
         return self.sdk.view_annotate_image(annotation_set_id, self.id)
+
+
+class AnnotatedImage(Image):
```
Comment (drewlr) [inline remo/domain/image.py:170]: I'm not sure about introducing AnnotatedImage, but if not we would have to think about a way to combine images with annotations when there are multiple annotation sets

**Review by drewlr — COMMENTED** [review]
We'd need to think about the interaction between the different domain objects (Dataset -> Image, and AnnotationSet -> Annotation)

**Comment by drewlr** [comment]
@vmanilo this looks good to me! I added some small tweaks, please have a look and if good we can merge

File: remo/domain/image.py (hunk 1 above)
Comment (vmanilo) [reply remo/domain/image.py:170]: ok, I can think about integrating Annotations into Image

Comment (drewlr) [reply remo/domain/image.py:170]: @vmanilo I think we can leave this for later, and merge for now. It requires understanding better the use cases, it's not something immediate I reckon

Comment (vmanilo) [reply remo/domain/image.py:170]: ok, sure

**Pull request merged**